be/src/olap/rowset/segment_v2/variant/variant_column_writer_impl.cpp

```diff
@@ -525,22 +655,27 @@ Status VariantColumnWriterImpl::_process_subcolumns(vectorized::ColumnVariant* p
     return Status::OK();
 }
 
-// Serialize and write Variant sparse data. Decides mode based on FE-configured bucket num,
-// initializes the corresponding writer(s), and delegates conversion/append/statistics to the
-// unified sparse writer. Column id consumption order remains identical to the previous logic.
 Status VariantColumnWriterImpl::_process_sparse_column(
         vectorized::ColumnVariant* ptr, vectorized::OlapBlockDataConvertor* converter,
         size_t num_rows, int& column_id) {
     int bucket_num = std::max(1, _tablet_column->variant_sparse_hash_shard_count());
-    if (bucket_num <= 1) {
-        TabletColumn sparse_column = vectorized::schema_util::create_sparse_column(*_tablet_column);
-        RETURN_IF_ERROR(_sparse_writer.init_single(sparse_column, column_id, _opts, _opts.footer));
-    } else {
-        RETURN_IF_ERROR(_sparse_writer.init_buckets(bucket_num, *_tablet_column, column_id, _opts,
-                                                    _opts.footer));
+    RETURN_IF_ERROR(
+            _sparse_writer.init(_tablet_column, bucket_num, column_id, _opts, _opts.footer));
+    RETURN_IF_ERROR(_sparse_writer.append_data(_tablet_column, *ptr, num_rows, converter));
+    return Status::OK();
+}
+
+Status VariantColumnWriterImpl::_process_doc_snapshot_column(
+        vectorized::ColumnVariant* ptr, vectorized::OlapBlockDataConvertor* converter,
+        size_t num_rows, int& column_id) {
+    if (!_tablet_column->variant_enable_doc_snapshot_mode()) {
+        return Status::OK();
     }
-    RETURN_IF_ERROR(_sparse_writer.append_from_variant(*ptr, num_rows, converter, *_tablet_column,
-                                                       &_statistics));
+    ptr->reconstruct_doc_snapshot_column();
```

Comment:
why need to `reconstruct_doc_snapshot_column` ? schema change ?  better add comment

Reply:
what if remove this logic?